src/ext/java/org/opentripplanner/ext/transmodelapi/model/plan/LegType.java

```diff
@@ -317,6 +319,23 @@ public static GraphQLObjectType create(
           })
           .build()
       )
+      .field(
+        GraphQLFieldDefinition
+          .newFieldDefinition()
+          .name("serviceDate")
+          .description(
+            "For transit legs, the service date of the trip. For non-transit legs, null."
+          )
+          .type(gqlUtil.localDateScalar)
```

Comment:
I think we should avoid adding new scalar types, as `Date` already exists. However it would be a good idea to convert it to use `LocalDate` internally